browser/net/url_context.cc

```diff
@@ -139,4 +154,56 @@ void BraveRequestInfo::FillCTXFromRequest(const net::URLRequest* request,
   ctx->upload_data = GetUploadDataFromURLRequest(request);
 }
 
+// static
+void BraveRequestInfo::FillCTX(
+    const network::ResourceRequest& request,
+    int render_process_id,
+    int frame_tree_node_id,
+    uint64_t request_identifier,
+    content::ResourceContext* resource_context,
+    std::shared_ptr<brave::BraveRequestInfo> ctx) {
+  ctx->request_identifier = request_identifier;
+  ctx->request_url = request.url;
+  // TODO(iefremov): Replace GURL with Origin
+  ctx->initiator_url =
+      request.request_initiator.value_or(url::Origin()).GetURL();
+
+  ctx->referrer = request.referrer;
+  ctx->referrer_policy = request.referrer_policy;
+
+  ctx->resource_type =
+      static_cast<content::ResourceType>(request.resource_type);
+
+  ctx->is_webtorrent_disabled = IsWebTorrentDisabled(resource_context);
+
+  ctx->render_frame_id = request.render_frame_id;
+  ctx->render_process_id = render_process_id;
+  ctx->frame_tree_node_id = frame_tree_node_id;
+
+  // TODO(iefremov): remove tab_url. Change tab_origin from GURL to Origin.
+  // ctx->tab_url = request.top_frame_origin;
+  ctx->tab_origin = request.top_frame_origin.value_or(url::Origin()).GetURL();
```

Comment:
`NetworkIsolationKey` will arrive only in C77

Reply:
I think we should do what I did in the other PR and only use `top_frame_origin` if `NetworkIsolationKey` is empty because I don't want this to be missed later

Reply:
atm there is no such field in `ResourceRequest`, so I can't use `NetworkIsolationKey`

Reply:
huh, I guess it's only in URLRequest

Reply:
It will come in C77, so we are safe
https://cs.chromium.org/chromium/src/services/network/public/cpp/resource_request.h?q=resourcerequest&sq=package:chromium&dr=CSs&l=44